aspnetcore/mvc/views/tag-helpers/builtin/resources/CacheTagHelper.md

```diff
@@ -0,0 +1,278 @@
+
+- - -
+
+### vary-by
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| String             | "@Model"  	            |
+
+
+The attribute `vary-by` allows for customization of what data gets cached.  When the string value changes of what is assigned to the attribute 'vary-by', the content of the `Cache` tag helper will be updated. Often a model value or combination of model values are assigned to this attribute.  Otherwise, the value will never change and the cache will be updated only by the inclusion of other attributes in the `Cache` tag helper.
+
+The example below assumes the controller method rendering the view sums the integer value of the two route parameters, myParam1 and myParam2, and returns that as the single model property. When this sum changes, the content of the `Cache` tag helper will be rendered again and the cache is reset.  
+
+Usage Example:
+
+
+Controller
+
+```C#
+public IActionResult Index
+    (string myParam1,string myParam2,string myParam3)
+{
+   int num1;
+    int num2;
+    int.TryParse(myParam1, out num1);
+    int.TryParse(myParam2, out num2);
+    return View(viewName, num1 + num2);
+}
+```
+
+Index.cshtml
+
+```HTML
+<Cache vary-by="@Model"">
+  Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+- - -
+
+### priority
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| CacheItemPriority  | "High"   	            |
+|                    | "Low" |
+|                    | "NeverRemove" |
+|                    | "Normal" |
+
+The attribute `priority` is used to give guidance to the built in cache provider on what priority to assign to the cache contents stored.  That is, there is no guarantee of the cache not being invalidated prior to the expected expiration.  If for example you set one `Cache` tag helper to have its priority `High` and another to `Low`, it is likely that if the web server experiences memory pressure and needs to evict one of your cached content values, it will evict the `Low` priority first.
+
+Usage Example:
+
+<Cache priority="High">
+  Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+
+> [!WARNING]
+>  using the `priority` attribute does not guarantee any specific level of cache retention. It is simply a suggestion to the Cache Provider. To be extra clear, setting this attribute to `NeverRemove` does not guarantee that the cache will not be evicted.  More details can be found in the additional resources listed below.
```

Comment:
@GuardRex Yours sounds better but the "NeverRemove" setting cost me a weekend once so I want to save someone else from the same fate.  My preference would be to call it "PleaseDontRemoveThisUnlessYouAbsolutelyHaveTo" but guessing that will not happen.  Any other suggestions for how to re-word?